src/els_dialyzer_diagnostics.erl

```diff
@@ -29,7 +29,8 @@ diagnostics(Uri) ->
   case els_config:get(plt_path) of
     undefined -> [];
     DialyzerPltPath ->
-      WS = try dialyzer:run([ {files, [binary_to_list(Path)]}
+      Dependencies = deps_paths(els_diagnostics_utils:dependencies(Uri)),
+      WS = try dialyzer:run([ {files, [binary_to_list(Path) | Dependencies]}
```

Comment:
Won't this end up analyzing all of the dependencies and also returning all the issues with all analyzed files? I don't think this is what we want, do we?

Reply:
It only looks for the dependencies of the particular module being analyzed. Surely this is more than only the one module but it does not appear to be much slower than it was before (disclaimer: no measurements just "feeling"). 

(Dialyzer runs done from Erlang LS do still misses some errors though but that's a different problem than the one I aim to fix here)

Reply:
Right, dependencies in this case are only the behaviours and/or parse transforms used.